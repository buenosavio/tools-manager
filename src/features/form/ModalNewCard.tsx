import { useFormik } from 'formik'
import { useDispatch } from 'react-redux'
import { toolActions } from '../../store/tools/duck'
import {
  Form,
  Modal,
  Container,
  TitleForm,
  Input,
  Label,
  TextArea,
  ButtonAdd,
  ButtonCancel,
  FooterForm
} from './ModalNewCard.styles'

interface OpenModal {
  setOpenModal: (arg0: boolean) => void
}

const ModalNewCard = ({ setOpenModal }: OpenModal) => {
  const dispatch = useDispatch()

  const formik = useFormik({
    initialValues: {
      name: '',
      description: '',
      link: '',
      tag: [],
      id: ''
    },
    onSubmit: values => {
      setOpenModal(false)
      dispatch(toolActions.create(values))
    }
  })

  return (
    <Container>
      <Modal>
        <Form onSubmit={formik.handleSubmit}>
          <TitleForm> + Add new tool</TitleForm>
          <Label htmlFor="name">Tool Name</Label>
          <Input
            type="text"
            name="name"
            id="name"
            onChange={formik.handleChange}
            value={formik.values.name}
          />
          <Label htmlFor="link">Tool Link</Label>
          <Input
            type="text"
            name="link"
            id="link"
            onChange={formik.handleChange}
            value={formik.values.link}
          />
          <Label htmlFor="description">Tool description</Label>
          <TextArea
            id="description"
            name="description"
            onChange={formik.handleChange}
            value={formik.values.description}
          />
          <Label htmlFor="tags">Tool Tags</Label>
          <Input
            type="text"
            id="tag"
            name="tag"
            onChange={formik.handleChange}
            value={formik.values.tag}
          />
          <FooterForm>
            <ButtonCancel
              onClick={() => {
                setOpenModal(false)
              }}
            >
              Cancel
            </ButtonCancel>
            <ButtonAdd type="submit">Add tool</ButtonAdd>
          </FooterForm>
        </Form>
      </Modal>
    </Container>
  )
}

export default ModalNewCard
