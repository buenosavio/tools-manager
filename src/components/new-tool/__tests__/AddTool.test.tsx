import ModalNewCard from '../AddTool'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import withProvider from '../../../utils/withContext'
import { mount } from 'enzyme'

describe('UI render should be default', () => {
  const initialState = { tools: [], isOnlyTag: true, loading: true, search: '' }
  const mockStore = configureStore()
  const store = mockStore(initialState)

  test('Snap of screen new modal should be renderized correctly', () => {
    const openModal = () => true
    const tree = renderer
      .create(
        <Provider store={store}>
          <ModalNewCard setOpenModal={openModal} />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('Form test values', () => {
  const initialState = { tools: [], isOnlyTag: true, loading: true, search: '' }
  const mockStore = configureStore()
  const store = mockStore(initialState)
  const openModal = () => true

  const Component = withProvider(ModalNewCard, store)

  test('input elements should be clickabed', () => {
    const renderComponent = mount(<Component setOpenModal={openModal} />)
    renderComponent.find('#tool-name').at(0).simulate('click')
  })
})
