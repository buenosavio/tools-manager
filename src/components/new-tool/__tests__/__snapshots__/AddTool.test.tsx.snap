// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI render should be default Snap of screen new modal should be renderized correctly 1`] = `
<div
  className="sc-bczRLJ fesoAM"
>
  <div
    className="sc-gsnTZi lpqPDs"
  >
    <form
      className="sc-dkzDqf jSJjnj"
    >
      <p
        className="sc-hKMtZM ddUFem"
      >
         + Add new tool
      </p>
      <label
        className="sc-eCYdqJ jvWRvA"
        htmlFor="tool-name"
      >
        Tool Name
      </label>
      <input
        className="sc-gKXOVf hYXwfN"
        id="tool-name"
        name="tool-name"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
      />
      <label
        className="sc-eCYdqJ jvWRvA"
        htmlFor="tool-link"
      >
        Tool Link
      </label>
      <input
        className="sc-gKXOVf hYXwfN"
        id="tool-link"
        name="tool-link"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
      />
      <label
        className="sc-eCYdqJ jvWRvA"
        htmlFor="tool-desc"
      >
        Tool Description
      </label>
      <textarea
        className="sc-jSMfEi eMJWXs"
        id="tool-desc"
        name="tool-desc"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
      />
      <footer
        className="sc-papXJ crZneE"
      >
        <button
          className="sc-iBkjds eDUIER"
          onClick={[Function]}
        >
          Cancel
        </button>
        <button
          className="sc-ftvSup rVEmU"
          type="submit"
        >
          Add tool
        </button>
      </footer>
    </form>
  </div>
</div>
`;
